spec/execution_tests.lua

```diff
@@ -1749,6 +1749,42 @@ function execution_tests.run(compile_file, backend, _ENV, only_compile)
         end)
     end)
 
+    describe("For-in loops", function()
+        compile([[
+            export function iter(a: any, b: any): (any, any, any)
+                return 1, 2, 3
+            end
+
+            export function fn()
+                local a: any = 2
+                local b: any = 3
+                for i, j, k in iter, a, b do
+                    i = j
+                    k = i
+                end
+            end
+        ]])
+    end)
+
+    describe("Nested for-in loops", function()
+        compile([[
```

Comment:
To speed up the test suite, we put related code in the same `compile` block. I'd put the test case for nested for-in loops in the same compile block as the other non-nested loop.